module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1310,6 +1310,20 @@ zfs_domount(vfs_t *vfsp, char *osname)
 
 	if (!zfsvfs->z_issnap)
 		zfsctl_create(zfsvfs);
+
+#ifdef __FreeBSD__
+	if (error == 0) {
+		/* zone dataset visiblity was checked before in zfs_mount() */
+		struct prison *pr = curthread->td_ucred->cr_prison;
+		if (pr != &prison0) {
+			zfsvfs->z_os->os_dsl_dataset->ds_jailname =
+			    kmem_zalloc(strlen(pr->pr_name) + 1, KM_SLEEP);
+			strcpy(zfsvfs->z_os->os_dsl_dataset->ds_jailname,
+			    pr->pr_name);
```

Comment:
You can use `kmem_strdup(pr->pr_name);` to simplify this.

Reply:
Indeed. Fixed.